server/container_create.go

```diff
@@ -623,6 +623,10 @@ func (s *Server) CreateContainer(ctx context.Context, req *pb.CreateContainerReq
 		log.Errorf(ctx, "%v", err)
 	}
 
+	if ctx.Err() == context.Canceled || ctx.Err() == context.DeadlineExceeded {
+		return nil, ctx.Err()
+	}
+
```

Comment:
is there any possibility that the request is cancelled after this check and before the response is received?

Reply:
Yeah that is possible but that should be okay since we end up cleaning the pod in CRI-O which is the main goal of this PR.
The issue earlier was we weren't cleaning up at all so kubelet crio were out of sync w.r.t. the pod state. Also, I think we need these checks sprinkled through the function more as a follow up.

Reply:
@giuseppe actually after rereading this comment you are right we still have that race. See my response to @mtrmac on thoughts to fix this via bumping up attempt number or on crio side adding a random string to name and doing cleanup on older leaked resources.